### PR TITLE
Rollback #1089 & #1141 unnecessary changes

### PR DESCRIPTION
**Pull request by AngelRodriguez8008**
### Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (updates to documentation, formatting, etc.)

### Description
This PR rollback unnecessary changes & add the related tests.
Add XrmApp.IsTabVisible(string tabName) method (small useful feature)

### Issues addressed
#1089 introduce a Bug, by adding a redundant condition. Please check comments in this PR 
#1141 Fix the Bug introduced in PR #1089
I retest this logic, in different environments with & without MFA, also deactivating the Keep Signed-In Dialog. It works in all cases.

### All submissions:

- [x] My code follows the code style of this project.
- [x] Do existing samples that are effected by this change still run?
- [x] I have added samples for new functionality. 
- [x] I raise detailed error messages when possible.
- [x] My code does not rely on labels that have the option to be hidden.

### Which browsers was this tested on?

- [x] Chrome
- [ ] Firefox
- [ ] IE
- [ ] Edge

### CODE REVIEW

File: Microsoft.Dynamics365.UIAutomation.Api/Pages/Entity.cs [hunk 1]
```diff
@@ -686,7 +686,7 @@ public BrowserCommandResult<bool> CollapseTab(string name, int thinkTime = Const
 
             return this.Execute(GetOptions($"Collapse Tab: {name}"), driver =>
             {
-                if (!driver.HasElement(By.XPath(Elements.Xpath[Reference.Entity.Tab].Replace("[NAME]", name))))
+                if (!IsTabVisible(driver, name))
```
Comment (AngelRodriguez8008) [inline Microsoft.Dynamics365.UIAutomation.Api/Pages/Entity.cs:689]: Small refactor

File: Microsoft.Dynamics365.UIAutomation.Api/Pages/Entity.cs [hunk 2]
```diff
@@ -1892,6 +1904,7 @@ public BrowserCommandResult<bool> SetHeaderValue(TwoOption option)
                             !optionValue && selectedValue == options.FirstOrDefault(a => a.GetAttribute("value") == "1")?.GetAttribute("title"))
                         {
                             driver.ClickWhenAvailable(By.XPath(Elements.Xpath[Reference.Entity.CheckboxFieldContainer_Header].Replace("[NAME]", option.Name.ToLower())));
+                            driver.ClearFocus();
```
Comment (AngelRodriguez8008) [inline Microsoft.Dynamics365.UIAutomation.Api/Pages/Entity.cs:1907]: Small fix

File: Microsoft.Dynamics365.UIAutomation.Sample/UCI/Login/Login.cs [hunk 3]
```diff
@@ -17,11 +17,6 @@ public class Login : TestsBase
         public void MultiFactorLogin()
         {
             _xrmApp.Grid.SwitchView("All Accounts");
-
-            _xrmApp.CommandBar.ClickCommand("New");
-
-            _xrmApp.Entity.SetValue("name", "Test API Account" + TestSettings.GetRandomString(5,5) );
-            _xrmApp.Entity.SetValue("telephone1", "555-555-5555");
         }
```
Comment (AngelRodriguez8008) [inline Microsoft.Dynamics365.UIAutomation.Sample/UCI/Login/Login.cs]: Remove unnecessary steps, not related to Login

**Comment by AngelRodriguez8008** [comment]
@imartinflores, @samadane, @tpalus, @ewingjm please consider review this PR, in case that you have some scenario to be revised, please contact me 

I will try to find a way to make the Login faster, eventually try reuse the authentication inside the test class.

**Comment by imartinflores** [comment]
Hi @AngelRodriguez8008 , the change was needed because we did not use MFA, and for some reason, the validation was going through and the error message "Something went wrong entering the OTC" was thrown. With the change, if you just set it as "",  it does work. I have not checked your PR, but if it is just a rollback , it probably causes the issue with the validation again. You can check the last Comments on the previous PR.

**Comment by imartinflores** [comment]
@AngelRodriguez8008 dont remember exactly now, but either the "EnterOneTimeCode" or "ClickStaySignedIn" were not evaluating property, then the "OTC message was thrown". Im guessing with the rollback, it will start happening again. As for now, that PR resolves it just by using MFA as an empty string .

**Comment by AngelRodriguez8008** [comment]
@imartinflores if you "resolves it just by using MFA as an empty string" than you condition  (mfaSecretKey != null) will be always true & is not required anyway.

I got this error too, in one scenario. I fixed it by Transform to Unsecure String and then comprare it if null or empty string.
```
private bool EnterOneTimeCode(IWebDriver driver, SecureString mfaSecretKey) {
    ...
    string key = mfaSecretKey?.ToUnsecureString(); // <- this 2FA secret key.
    if (string.IsNullOrWhiteSpace(key))
           throw new InvalidOperationException("The application is wait for the OTC but your MFA-SecretKey is not set. Please check your configuration.");
    ...
```
Thanks for your replay

**Comment by imartinflores** [comment]
@AngelRodriguez8008 will be true if you specify some value, even an empty string. If you do not specify anything, it will be false, and will go through the validation. Anyways, happy to hear you've done that as you could reproduce the same issue seems like. So, I guess all sorted :)

**Review by TYLEROL — APPROVED** [review]

**Pull request merged**